Packs/Base/Scripts/DBotMLFetchData/DBotMLFetchData.py

```diff
@@ -1,6 +1,1031 @@
+    lines = (line.strip() for line in text.splitlines())
+    # break multi-headlines into a line each
+    chunks = (phrase.strip() for line in lines for phrase in line.split("  "))
+    # drop blank lines
+    text = '\n'.join(chunk for chunk in chunks if chunk)
+    return text
+
+
+def get_vocab_features(text, ngrams_counter):
+    global word_to_ngram, word_to_regex
+    vocav_features = {}
+    for word, regex in word_to_regex.items():
+        count = len(re.findall(regex, text))
+        if count > 0:
+            vocav_features[word] = count
+    for word, ngram in word_to_ngram.items():
+        count = ngrams_counter[ngram] if ngram in ngrams_counter else 0
+        if count > 0:
+            vocav_features[word] = count
+    return vocav_features
+
+
+def get_lexical_features(email_subject, email_body, email_body_word_tokenized, email_subject_word_tokenized):
+    words = [w for w in email_body_word_tokenized if any(c.isalnum() for c in w)]
+    num_of_words = len(words)
+    avg_word_length = 0 if len(words) == 0 else float(sum([len(w) for w in words])) / len(words)
+    sentences = sent_tokenize(email_body)
+    num_of_sentences = len(sentences)
+    avg_sentence_length = 0 if len(sentences) == 0 else float(sum([len(s) for s in sentences])) / len(sentences)
+    text_length = len(email_body)
+    subject_length = len(email_subject)
+    number_of_lines = email_body.count('\n') + 1
+    word_subject = [w for w in email_subject_word_tokenized if any(c.isalnum() for c in w)]
+    num_of_words_subject = len(word_subject)
+    ending_dots = len([s for s in sentences if s.strip().endswith('.')])
+    ending_explanation = len([s for s in sentences if s.strip().endswith('!')])
+    ending_question = len([s for s in sentences if s.strip().endswith('?')])
+
+    return {'num_of_words': num_of_words,
+            'avg_word_length': avg_word_length,
+            'num_of_sentences': num_of_sentences,
+            'avg_sentence_length': avg_sentence_length,
+            'avg_number_of_words_in_sentence': float(num_of_words) / num_of_sentences if num_of_sentences > 0 else 0,
+            'text_length': text_length,
+            'num_of_words_subject': num_of_words_subject,
+            'subject_length': subject_length,
+            'ending_dots': ending_dots,
+            'ending_explanation': ending_explanation,
+            'ending_question': ending_question,
+            'number_of_lines': number_of_lines
+            }
+
+
+def get_characters_features(text):
+    characters_count = {}
+    for c in string.printable:
+        if c in ['[', ']', '<']:
+            continue
+        count = text.count(c)
+        if count > 0:
+            characters_count[c] = count
+    return characters_count
+
+
+def get_url_features(email_body, email_html, soup):
+    http_regex = r'http://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
+    https_regex = r'https://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
```

Comment:
if you stay with your regex (might be easier) please add support with `hxxp` it's proofpoint format